packages/trackerless-network/src/NetworkStack.ts

```diff
@@ -119,10 +132,18 @@ export class NetworkStack {
         return this.layer0Node!
     }
 
+    getConnectionManager(): ConnectionManager {
+        return this.layer0Node!.getTransport() as ConnectionManager
+    }
+
     getMetricsContext(): MetricsContext {
         return this.metricsContext
     }
 
+    async fetchNodeInfo(node: PeerDescriptor, getControlLayerInfo: boolean, getStreamPartitionInfo: StreamPartID[]): Promise<InfoResponse> {
+        return this.infoClient!.getInfo(node, getControlLayerInfo, getStreamPartitionInfo)
```

Comment:
Could inline `InfoClient` class and use the `InfoRpcRemote` here directly?

```suggestion
        const rpcCommunicator = new RoutingRpcCommunicator(...)
        const remote = new InfoRpcRemote(this.ownPeerDescriptor, node, rpcCommunicator, InfoRpcClient)
        return remote.getInfo(getControlLayerInfo, streamParts)
```

Reply:
Don't really want to create a new RpcCommunicator for each individual request

Reply:
Is the creation of `RpcCommunicator` a slow operation? If I understood correctly, we'd execute fetchNodeInfo relatively rarely (e.g. once a day for a node), and therefore it wouldn't be a problem if the operation would take e.g. 100 ms. It would simplify the state and be more descriptive as InfoRpcRemote doesn't need the listening functionality of a `ListeningRpcCommunicator`.

But if the creation of `RpcCommunicator` takes too much time, could do the inlining also without it.

Reply:
creating the RpcCommunicators is not really too heavy but it makes more sense to simply create a single RpcCommunicator for all requests that are done on the info rpc. The alternative is to create a new RpcCommunicator for each request which does not really make sense in the long run.

Reply:
Let's keep it as is. We couldn't arrive at a common outlook.